### PR TITLE
BUG: Fix `Timestamp('now')` and `Timestamp.now` unit inconsistency

### LINKED ISSUE #55879: BUG: variable datetime dtype since Pandas 2.1.0
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame()

df['date1'] = pd.Timestamp.now()
df['date2'] = pd.Timestamp('now')
df['date3'] = pd.Timestamp('2023-11-08 11:13:14')

df.dtypes
```

### Issue Description

Three columns above have three different types:
```
date1    datetime64[us]
date2    datetime64[ns]
date3     datetime64[s]
```

This is rather unexpected behavior. I could not find an explanation for it on https://pandas.pydata.org/docs/dev/whatsnew/v2.1.0.html#

### Expected Behavior

```
date1    datetime64[ns]
date2    datetime64[ns]
date3    datetime64[ns]```

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit              : a60ad39b4a9febdea9a59d602dad44b1538b0ea5
python              : 3.11.5.final.0
python-bits         : 64
OS                  : Linux
OS-release          : 5.10.102.1-microsoft-standard-WSL2
Version             : #1 SMP Wed Mar 2 00:30:59 UTC 2022
machine             : x86_64
processor           : x86_64
byteorder           : little
LC_ALL              : en_US.UTF-8
LANG                : en_US.UTF-8
LOCALE              : en_US.UTF-8

pandas              : 2.1.2
numpy               : 1.24.2
pytz                : 2023.3.post1
dateutil            : 2.8.2
setuptools          : 65.5.0
pip                 : 23.3.1
Cython              : None
pytest              : None
hypothesis          : None
sphinx              : None
blosc               : None
feather             : None
xlsxwriter          : None
lxml.etree          : None
html5lib            : None
pymysql             : None
psycopg2            : 2.9.6
jinja2              : None
IPython             : 8.16.1
pandas_datareader   : None
bs4                 : None
bottleneck          : None
dataframe-api-compat: None
fastparquet         : None
fsspec              : None
gcsfs               : None
matplotlib          : 3.8.0
numba               : None
numexpr             : None
odfpy               : None
openpyxl            : None
pandas_gbq          : None
pyarrow             : None
pyreadstat          : None
pyxlsb              : None
s3fs                : None
scipy               : None
sqlalchemy          : 1.4.49
tables              : None
tabulate            : None
xarray              : None
xlrd                : None
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None
</details>

### PR DESCRIPTION
**Pull request by yuanx749**
This fix make `Timestamp('now')` and `Timestamp('today')` return a Timestamp with unit 'us', the same as `Timestamp.now()` and `Timestamp.today()`.

- [x] closes #55879(Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/conversion.pyx [hunk 1]
```diff
@@ -646,7 +646,7 @@ cdef _TSObject convert_str_to_tsobject(str ts, tzinfo tz,
         reso = get_supported_reso(out_bestunit)
         return convert_datetime_to_tsobject(dt, tz, nanos=nanos, reso=reso)
 
-    return convert_datetime_to_tsobject(dt, tz)
+    return convert_datetime_to_tsobject(dt, tz, nanos=0, reso=NPY_FR_us)
```
Comment (mroeschke) [inline pandas/_libs/tslibs/conversion.pyx:649]: This general case probably cannot be hardcoded

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/conversion.pyx (hunk 1 above)
Comment (yuanx749) [reply pandas/_libs/tslibs/conversion.pyx:649]: Looks like the only cases left to handle here are 'now' and 'today', all other branches have returned.

I think we could combine these two cases and then return, to make the code more clear.
https://github.com/pandas-dev/pandas/blob/d6fdbee9f31f621f5177f65abc4525c822ab2e97/pandas/_libs/tslibs/conversion.pyx#L597-L605

File: doc/source/whatsnew/v2.1.4.rst [hunk 2]
```diff
@@ -22,6 +22,7 @@ Fixed regressions
 Bug fixes
 ~~~~~~~~~
 - Bug in :class:`Series` constructor raising DeprecationWarning when ``index`` is a list of :class:`Series` (:issue:`55228`)
+- Bug in :class:`Timestamp` construction with ``ts_input=now`` or ``ts_input=today`` giving a different unit from :meth:`Timestamp.now` or :meth:`Timestamp.today` (:issue:`55879`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.4.rst:25]:
```suggestion
- Bug in :class:`Timestamp` construction with ``ts_input="now"`` or ``ts_input="today"`` giving a different unit from :meth:`Timestamp.now` or :meth:`Timestamp.today` (:issue:`55879`)
```

File: pandas/tests/scalar/timestamp/test_constructors.py [hunk 3]
```diff
@@ -464,6 +464,20 @@ def test_constructor_str_infer_reso(self):
         ts = Timestamp("2020-01-01 00+00:00")
         assert ts.unit == "s"
 
+    def test_now_today_unit(self):
+        # GH#55879
+        ts_now_from_method = Timestamp.now()
```
Comment (mroeschke) [inline pandas/tests/scalar/timestamp/test_constructors.py:469]: Can you use pytest.mark.parametrize?

Comment (yuanx749) [reply pandas/tests/scalar/timestamp/test_constructors.py]: Done.

File: pandas/_libs/tslibs/conversion.pyx [hunk 4]
```diff
@@ -594,15 +594,12 @@ cdef _TSObject convert_str_to_tsobject(str ts, tzinfo tz,
...
+    elif ts in ("now", "today"):
         # Issue 9000, we short-circuit rather than going
         # into np_datetime_strings which returns utc
         dt = datetime.now(tz)
-    elif ts == "today":
-        # Issue 9000, we short-circuit rather than going
-        # into np_datetime_strings which returns a normalized datetime
-        dt = datetime.now(tz)
         # equiv: datetime.today().replace(tzinfo=tz)
+        return convert_datetime_to_tsobject(dt, tz, nanos=0, reso=NPY_FR_us)
```
[the remark this answers is not in the document]
Comment (yuanx749) [reply pandas/_libs/tslibs/conversion.pyx:602]: No. I removed that line.

**Review by mroeschke — APPROVED** [review]
LGTM cc @jbrockmendel

File: pandas/_libs/tslibs/conversion.pyx [hunk 5]
```diff
@@ -594,15 +594,12 @@ cdef _TSObject convert_str_to_tsobject(str ts, tzinfo tz,
         obj.value = NPY_NAT
         obj.tzinfo = tz
         return obj
-    elif ts == "now":
+    elif ts in ("now", "today"):
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/conversion.pyx:597]: i think ATM we don't use a tuple here for perf reasons.  unless something has changed, cython constructs a tuple object

Comment (yuanx749) [reply pandas/_libs/tslibs/conversion.pyx]: I see. Changed back in the new commit.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @yuanx749

**Pull request merged**